File: R5ProObjectiveCExamples/R5ProObjectiveCExamples/Tests/Publish/PublishTest.m
//
//  PublishTest.m
//  R5ProObjectiveCExamples
//
//  Created by David Heimann on 6/6/17.
//  Copyright © 2017 Infrared5. All rights reserved.
//

#import "PublishTest.h"
#import "Testbed.h"

@interface PublishTest ()

@end

@implementation PublishTest

-(void) viewDidAppear:(BOOL)animated{
    
    [super viewDidAppear:animated];
    
    [[AVAudioSession sharedInstance] requestRecordPermission:^(BOOL granted) {
        
    }];
    
    [self setupDefaultR5VideoViewController];
    
    R5Configuration* config = [self getConfig];
    R5Connection* connection = [[R5Connection alloc] initWithConfig:config];
    
    [self setupPublisher:connection];
    
    [self.currentView attachStream:self.publishStream];
    
    [self.publishStream publish:[Testbed getParameter:@"stream1"] type:R5RecordTypeRecord];
}

@end

File: R5ProTestbed/Tests/Publish/PublishTest.swift
//
//  PublishTest.swift
//  R5ProTestbed
//
//  Created by Andy Zupko on 12/16/15.
//  Copyright © 2015 Infrared5. All rights reserved.
//

import UIKit
import R5Streaming

@objc(PublishTest)
class PublishTest: BaseTest {

    
    override func viewDidAppear(animated: Bool) {

        super.viewDidAppear(animated)
        
        AVAudioSession.sharedInstance().requestRecordPermission { (gotPerm: Bool) -> Void in
           
        };
        
        
        setupDefaultR5VideoViewController()
        
        // Set up the configuration
        let config = getConfig()
        // Set up the connection and stream
        let connection = R5Connection(config: config)
        
        setupPublisher(connection)
        // show preview and debug info
       // self.publishStream?.getVideoSource().fps = 2;
        self.currentView!.attachStream(publishStream!)
        
        
        self.publishStream!.publish(Testbed.getParameter("stream1") as! String, type: R5RecordTypeLive)
        


    }
}
